File: doc/examples/fetch_dbt.py
#!/usr/bin/env python3
"""Fetch verse text from DBT (Bible Brain, Digital Bible Platform, by
Faith Comes By Hearing) - requires your own API key (this repo doesn't
proxy DBT access). Get one at https://www.faithcomesbyhearing.com/bible-brain

No dependencies beyond the Python standard library.

Usage:
    BIBLE_API_KEY=... python3 fetch_dbt.py <fileset_id> <BOOK> <chapter>
    BIBLE_API_KEY=... python3 fetch_dbt.py ENGESVN2DA REV 15
"""
import json
import os
import sys
import urllib.request

API_BASE = "https://4.dbt.io/api"


def main():
    if len(sys.argv) < 4:
        sys.exit("usage: fetch_dbt.py <fileset_id> <BOOK> <chapter>")
    api_key = os.environ.get("BIBLE_API_KEY")
    if not api_key:
        sys.exit("Set BIBLE_API_KEY first - get one at "
                  "https://www.faithcomesbyhearing.com/bible-brain")

    fileset_id, book, chapter = sys.argv[1], sys.argv[2], sys.argv[3]
    url = f"{API_BASE}/bibles/filesets/{fileset_id}/{book}/{chapter}?key={api_key}&v=4"
    with urllib.request.urlopen(url) as r:
        data = json.loads(r.read())

    verses = data.get("data") or []
    if verses and "path" in verses[0]:
        print("This fileset returns a downloadable file, not inline verses:")
        print(verses[0]["path"])
        return

    for verse in verses:
        print(f"{verse['verse_start']}. {verse['verse_text']}")


if __name__ == "__main__":
    main()

File: doc/examples/fetch_dbt.js
#!/usr/bin/env node
// Fetch verse text from DBT (Bible Brain, Digital Bible Platform, by
// Faith Comes By Hearing) - requires your own API key (this repo doesn't
// proxy DBT access). Get one at https://www.faithcomesbyhearing.com/bible-brain
//
// No dependencies - uses the built-in fetch() (Node 18+, or any browser).
//
// Usage: BIBLE_API_KEY=... node fetch_dbt.js <fileset_id> <BOOK> <chapter>

const API_BASE = "https://4.dbt.io/api";

async function main() {
  const [filesetId, book, chapter] = process.argv.slice(2);
  const apiKey = process.env.BIBLE_API_KEY;
  if (!filesetId || !book || !chapter) {
    console.error("usage: BIBLE_API_KEY=... node fetch_dbt.js <fileset_id> <BOOK> <chapter>");
    process.exit(1);
  }
  if (!apiKey) {
    console.error(
      "Set BIBLE_API_KEY first - get one at https://www.faithcomesbyhearing.com/bible-brain"
    );
    process.exit(1);
  }

  const url = `${API_BASE}/bibles/filesets/${filesetId}/${book}/${chapter}?key=${apiKey}&v=4`;
  const data = await (await fetch(url)).json();
  const verses = data.data || [];

  if (verses.length && verses[0].path) {
    console.log("This fileset returns a downloadable file, not inline verses:");
    console.log(verses[0].path);
    return;
  }

  for (const verse of verses) {
    console.log(`${verse.verse_start}. ${verse.verse_text}`);
  }
}

main();
